core/src/test/kotlin/org/stellar/anchor/sep24/Sep24ServiceTest.kt

```diff
@@ -270,6 +271,26 @@ internal class Sep24ServiceTest {
     assertEquals(withdrawQuote.buyAsset, slotTxn.captured.amountOutAsset)
   }
 
+  @Test
+  fun `test withdraw with user_action_required_by`() {
+    val slotTxn = slot<Sep24Transaction>()
+    val now = Instant.now()
+    val deadline = 100L
+    every { txnStore.save(capture(slotTxn)) } returns null
+    every { sep24Config.initialUserDeadlineSeconds } returns deadline
+    sep24Service.withdraw(createTestSep10JwtWithMemo(), createTestTransactionRequest())
+    val dbDeadline = slotTxn.captured.userActionRequiredBy.epochSecond
+    val expectedDeadline = Instant.now().plusSeconds(deadline).epochSecond
```

Comment:
We really should be injecting a Clock everywhere 😢

Reply:
Created a task for that
https://stellarorg.atlassian.net/browse/ANCHOR-705